api/src/main/java/org/pmiops/workbench/interactiveanalysis/InteractiveAnalysisService.java

```diff
@@ -0,0 +1,203 @@
+    // in workspace IDs.
+    String gcsNotebooksDir = "gs://" + firecloudWorkspace.getBucketName() + "/notebooks";
+    long workspaceId = dbWorkspace.getWorkspaceId();
+
+    fileNames.forEach(
+        notebookName ->
+            userRecentResourceService.updateNotebookEntry(
+                workspaceId, userProvider.get().getUserId(), gcsNotebooksDir + "/" + notebookName));
+
+    String workspacePath = dbWorkspace.getFirecloudName();
+    String googleProjectId = dbWorkspace.getGoogleProject();
+    // Use current dir if not Jupyter
+    String editDir = isGceRuntime ? "workspaces/" + workspacePath : "";
+    String playgroundDir =
+        isGceRuntime ? "workspaces_playground/" + workspacePath : "workspaces_playground";
+    String targetDir = isPlayground ? playgroundDir : editDir;
+
+    if (isGceRuntime) {
+      leonardoNotebooksClient.createStorageLinkForRuntime(
+          googleProjectId,
+          appName,
+          new StorageLink()
+              .cloudStorageDirectory(gcsNotebooksDir)
+              .localBaseDirectory(editDir)
+              .localSafeModeBaseDirectory(playgroundDir)
+              .pattern(JUPYTER_DELOC_PATTERN));
+    } else {
+      // For now if the request is not for GCE runtime, that would be RStudio. When supporting more
+      // apps, consider to use AppType.
+      leonardoNotebooksClient.createStorageLinkForApp(
+          googleProjectId,
+          appName,
+          new StorageLink()
+              .cloudStorageDirectory(gcsNotebooksDir)
+              .localBaseDirectory(editDir)
+              .localSafeModeBaseDirectory(playgroundDir)
+              .pattern(RSTUDIO_DELOC_PATTERN));
+    }
+
+    // Always localize config files; usually a no-op after the first call.
+    Map<String, String> localizeMap = new HashMap<>();
+
+    // The Welder extension offers direct links to/from playground mode; write the AoU config file
+    // to both locations so notebooks will work in either directory.
+
+    String aouConfigUri =
+        aouConfigDataUri(
+            firecloudWorkspace,
+            cdrVersion,
+            workspaceNamespace,
+            workbenchConfigProvider.get().server.apiBaseUrl);
+
+    // If current dir is empty (e.g. creating RStudio notebook), don't append '/', otherwise, it
+    // becomes root directory.
+    String aouConfigEditDir =
+        editDir.isEmpty() ? AOU_CONFIG_FILENAME : editDir + "/" + AOU_CONFIG_FILENAME;
+    String localizeTargetDir = targetDir.isEmpty() ? "" : targetDir + "/";
+
+    localizeMap.put(aouConfigEditDir, aouConfigUri);
+    localizeMap.put(playgroundDir + "/" + AOU_CONFIG_FILENAME, aouConfigUri);
+
+    // Localize the requested notebooks, if any.
+    localizeMap.putAll(
+        fileNames.stream()
+            .collect(
+                Collectors.toMap(
+                    name -> localizeTargetDir + name, name -> gcsNotebooksDir + "/" + name)));
+    log.info(localizeMap.toString());
```

Comment:
Just curious We need to log this info?

Reply:
It was added https://github.com/all-of-us/workbench/blob/0eeaa6f81a946a4bd3234f4a2b2a7b9b51bc7ea6/api/src/main/java/org/pmiops/workbench/api/RuntimeController.java#L370, I just copied it over. But I agree this is not useful. I will remove it. 
Thanks!